rpc/namespaces/ethereum/eth/filters/filter_system.go

```diff
@@ -101,49 +70,97 @@ func (es *EventSystem) subscribe(sub *Subscription) (*Subscription, pubsub.Unsub
 	)
 
 	ctx, cancelFn := context.WithCancel(context.Background())
-	defer cancelFn()
-
-	existingSubs := es.eventBus.Topics()
-	for _, topic := range existingSubs {
-		if topic == sub.event {
-			eventCh, unsubFn, err := es.eventBus.Subscribe(sub.event)
-			if err != nil {
-				err := errors.Wrapf(err, "failed to subscribe to topic: %s", sub.event)
-				return nil, nil, err
-			}
-
-			sub.eventCh = eventCh
-			return sub, unsubFn, nil
-		}
-	}
-
+	var query string
 	switch sub.typ {
-	case filters.LogsSubscription:
-		err = es.tmWSClient.Subscribe(ctx, sub.event)
-	case filters.BlocksSubscription:
-		err = es.tmWSClient.Subscribe(ctx, sub.event)
-	case filters.PendingTransactionsSubscription:
-		err = es.tmWSClient.Subscribe(ctx, sub.event)
+	case filters.LogsSubscription, filters.BlocksSubscription, filters.PendingTransactionsSubscription:
+		query = sub.event
 	default:
 		err = fmt.Errorf("invalid filter subscription type %d", sub.typ)
 	}
 
-	if err != nil {
-		sub.err <- err
-		return nil, nil, err
+	filter := coretypes.EventFilter{Query: query}
+	defaultFilter := coretypes.EventFilter{Query: ""}
+	defaultMaxItems := 1
+	if sub.after == defaultAfter {
+		res, resErr := es.client.Events(ctx, &coretypes.RequestEvents{
```

Comment:
Ideally job could be done within single call when tm support extra earliest or latest param like kafka on 1st call.